Libs/MRML/DisplayableManager/vtkMRMLSliceIntersectionInteractionRepresentation.cxx

```diff
@@ -0,0 +1,2333 @@
+    {
+    vtkNew<vtkMatrix3x3> rotationMatrix;
+    vtkNew<vtkMatrix3x3> identityMatrix;
+    vtkNew<vtkMatrix3x3> kmat;
+    kmat->SetElement(0, 0, 0.0);
+    kmat->SetElement(0, 1, -v[2]);
+    kmat->SetElement(0, 2, v[1]);
+    kmat->SetElement(1, 0, v[2]);
+    kmat->SetElement(1, 1, 0.0);
+    kmat->SetElement(1, 2, -v[0]);
+    kmat->SetElement(2, 0, -v[1]);
+    kmat->SetElement(2, 1, v[0]);
+    kmat->SetElement(2, 2, 0.0);
+    vtkNew<vtkMatrix3x3> kmat2;
+    vtkMatrix3x3::Multiply3x3(kmat, kmat, kmat2);
+    vtkNew<vtkMatrix3x3> kmat2x;
+    rotationMatrix->SetElement(0, 0, identityMatrix->GetElement(0, 0) + kmat->GetElement(0, 0) + kmat2->GetElement(0, 0) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(0, 1, identityMatrix->GetElement(0, 1) + kmat->GetElement(0, 1) + kmat2->GetElement(0, 1) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(0, 2, identityMatrix->GetElement(0, 2) + kmat->GetElement(0, 2) + kmat2->GetElement(0, 2) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(1, 0, identityMatrix->GetElement(1, 0) + kmat->GetElement(1, 0) + kmat2->GetElement(1, 0) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(1, 1, identityMatrix->GetElement(1, 1) + kmat->GetElement(1, 1) + kmat2->GetElement(1, 1) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(1, 2, identityMatrix->GetElement(1, 2) + kmat->GetElement(1, 2) + kmat2->GetElement(1, 2) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(2, 0, identityMatrix->GetElement(2, 0) + kmat->GetElement(2, 0) + kmat2->GetElement(2, 0) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(2, 1, identityMatrix->GetElement(2, 1) + kmat->GetElement(2, 1) + kmat2->GetElement(2, 1) * ((1 - c) / (pow(s, 2.0))));
+    rotationMatrix->SetElement(2, 2, identityMatrix->GetElement(2, 2) + kmat->GetElement(2, 2) + kmat2->GetElement(2, 2) * ((1 - c) / (pow(s, 2.0))));
+
+    // Convert to 4x4 matrix
+    rotationMatrixHom->SetElement(0, 0, rotationMatrix->GetElement(0, 0));
+    rotationMatrixHom->SetElement(0, 1, rotationMatrix->GetElement(0, 1));
+    rotationMatrixHom->SetElement(0, 2, rotationMatrix->GetElement(0, 2));
+    rotationMatrixHom->SetElement(0, 3, 0.0);
+    rotationMatrixHom->SetElement(1, 0, rotationMatrix->GetElement(1, 0));
+    rotationMatrixHom->SetElement(1, 1, rotationMatrix->GetElement(1, 1));
+    rotationMatrixHom->SetElement(1, 2, rotationMatrix->GetElement(1, 2));
+    rotationMatrixHom->SetElement(1, 3, 0.0);
+    rotationMatrixHom->SetElement(2, 0, rotationMatrix->GetElement(2, 0));
+    rotationMatrixHom->SetElement(2, 1, rotationMatrix->GetElement(2, 1));
+    rotationMatrixHom->SetElement(2, 2, rotationMatrix->GetElement(2, 2));
+    rotationMatrixHom->SetElement(2, 3, 0.0);
+    rotationMatrixHom->SetElement(3, 0, 0.0);
+    rotationMatrixHom->SetElement(3, 1, 0.0);
+    rotationMatrixHom->SetElement(3, 2, 0.0);
+    rotationMatrixHom->SetElement(3, 3, 1.0);
+    }
+}
```

Comment:
The vtkMRMLSliceIntersectionInteractionRepresentation.cxx file is too big. It is not reasonable to have a 114KB file for implementing this, but if it turns out to be really that complex then at least we need to spin out some parts of it to helper class(es) to make complexity manageable.